### PR TITLE
Improving the ReadMe

### PR DESCRIPTION
**Pull request by quac88**
This PR fixes most of the python and adds more notes for the reader. I was not able to fix "3.3. Validator" and "4.8 Querying others in the network." In addition, ``loss.backward() # Accumulate gradients on endpoints`` remains broken on "3.1. Client"

### CODE REVIEW

**Comment by quac88** [comment]
Rebase of #1068

**Review by unconst — APPROVED** [review]
